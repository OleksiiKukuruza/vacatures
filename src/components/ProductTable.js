import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Cell, Row } from './Table';

const StyledProductTable = styled.div`
  display: flex;
  flex-direction: column;
  border-right: none;
`;

export const StyledRow = styled(Row)`
  cursor: pointer;
  background-color: ${({ active, theme }) => active ? theme.background : theme.primary.contrastText};
`;

const ProductTable = ({ vacancies, selectedVacancyId, selectVacancy }) => (
  <StyledProductTable>
    <Row>
      <Cell size={6}>
        Abonnement
      </Cell>
      <Cell size={2}>
        Periode
      </Cell>
      <Cell size={2}>
        Normale Prijs
      </Cell>
      <Cell size={2}>
        Prijs
      </Cell>
    </Row>
    {vacancies.map(vacancy => (
      <StyledRow
        key={vacancy.id}
        onClick={() => selectVacancy(vacancy.id)}
        active={selectedVacancyId === vacancy.id}
      >
        <Cell size={6}>
          <input
            type="radio"
            readOnly
            checked={selectedVacancyId === vacancy.id}
          />
          {vacancy.name}
        </Cell>
        <Cell size={2}>
          {vacancy.period}
        </Cell>
        <Cell size={2}>
          {vacancy.originalPrice}
        </Cell>
        <Cell size={2}>
          {vacancy.originalPrice - vacancy.discount}
        </Cell>
      </StyledRow>
    ))}
  </StyledProductTable>
);

ProductTable.propTypes = {
  vacancies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    period: PropTypes.number.isRequired,
    originalPrice: PropTypes.number.isRequired,
    discount: PropTypes.number.isRequired
  })).isRequired,
  selectedVacancyId: PropTypes.string,
  selectVacancy: PropTypes.func.isRequired
};

export default ProductTable;
