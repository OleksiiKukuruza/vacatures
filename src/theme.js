const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
