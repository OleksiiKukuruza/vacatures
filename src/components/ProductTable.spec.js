import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import ProductTable, { StyledRow } from './ProductTable';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('ProductTable', () => {
  it('renders without crashing', () => {
    const vacancies = [{
      id: '1',
      name: 'test',
      period: 21,
      originalPrice: 50,
      discount: 10
    }, {
      id: '2',
      name: 'test1',
      period: 32,
      originalPrice: 60,
      discount: 20
    }];
    const props = { vacancies, selectVacancy: () => null, selectedVacancyId: '1' };
    const { wrapper } = setup(ProductTable, props);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles click on a row', () => {
    const vacancies = [{
      id: '1',
      name: 'test',
      period: 21,
      originalPrice: 50,
      discount: 10
    }, {
      id: '2',
      name: 'test1',
      period: 32,
      originalPrice: 60,
      discount: 20
    }];
    const selectVacancy = jest.fn();
    const props = { vacancies, selectVacancy, selectedVacancyId: '1' };
    const { wrapper } = setup(ProductTable, props);
    const firstRow = wrapper.find(StyledRow).at(0);
    firstRow.simulate('click');
    expect(selectVacancy).toHaveBeenCalledWith(props.selectedVacancyId);
  });

  it('renders StyledRow in active state', () => {
    const { wrapper } = setup(StyledRow, { active: true, theme });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledRow in passive state', () => {
    const { wrapper } = setup(StyledRow, { active: false, theme });
    expect(wrapper).toMatchSnapshot();
  });
});
