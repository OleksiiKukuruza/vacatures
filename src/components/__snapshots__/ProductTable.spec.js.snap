// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable renders StyledRow in active state 1`] = `
.c0 {
  cursor: pointer;
  background-color: #f1f4f5;
}

<styled.div
  active={true}
  className="c0"
  theme={
    Object {
      "background": "#f1f4f5",
      "error": "#f44336",
      "primary": Object {
        "contrastText": "#fff",
        "main": "#54b6c8",
      },
      "spacingUnit": 8,
    }
  }
/>
`;

exports[`ProductTable renders StyledRow in passive state 1`] = `
.c0 {
  cursor: pointer;
  background-color: #fff;
}

<styled.div
  active={false}
  className="c0"
  theme={
    Object {
      "background": "#f1f4f5",
      "error": "#f44336",
      "primary": Object {
        "contrastText": "#fff",
        "main": "#54b6c8",
      },
      "spacingUnit": 8,
    }
  }
/>
`;

exports[`ProductTable renders without crashing 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      size={6}
    >
      Abonnement
    </styled.div>
    <styled.div
      size={2}
    >
      Periode
    </styled.div>
    <styled.div
      size={2}
    >
      Normale Prijs
    </styled.div>
    <styled.div
      size={2}
    >
      Prijs
    </styled.div>
  </styled.div>
  <Styled(styled.div)
    active={true}
    key="1"
    onClick={[Function]}
  >
    <styled.div
      size={6}
    >
      <input
        checked={true}
        readOnly={true}
        type="radio"
      />
      test
    </styled.div>
    <styled.div
      size={2}
    >
      21
    </styled.div>
    <styled.div
      size={2}
    >
      50
    </styled.div>
    <styled.div
      size={2}
    >
      40
    </styled.div>
  </Styled(styled.div)>
  <Styled(styled.div)
    active={false}
    key="2"
    onClick={[Function]}
  >
    <styled.div
      size={6}
    >
      <input
        checked={false}
        readOnly={true}
        type="radio"
      />
      test1
    </styled.div>
    <styled.div
      size={2}
    >
      32
    </styled.div>
    <styled.div
      size={2}
    >
      60
    </styled.div>
    <styled.div
      size={2}
    >
      40
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;
